src/main/java/org/apache/sysml/runtime/instructions/cp/ParamservBuiltinCPInstruction.java

```diff
@@ -381,26 +405,17 @@ private void doDataPartition(ExecutionContext ec, List<LocalPSWorker> workers) {
 				throw new DMLRuntimeException(String.format("Paramserv function: not support data partition scheme '%s'", getParam(PS_SCHEME)));
 			}
 		}
-		switch (scheme) {
-		case DISJOINT_CONTIGUOUS:
-			disjointContiguous(features, labels, valFeatures, valLabels, workers);
-			break;
-		case DISJOINT_RANDOM:
-		case OVERLAP_RESHUFFLE:
-		case DISJOINT_ROUND_ROBIN:
-			throw new DMLRuntimeException(String.format("Paramserv function: the scheme '%s' is not supported.", scheme));
-		}
+		return scheme;
 	}
 
-	private void disjointContiguous(MatrixObject features, MatrixObject labels, MatrixObject valFeatures,
+	private void doDataPartition(DataPartitioner dp, MatrixObject features, MatrixObject labels, MatrixObject valFeatures,
 			MatrixObject valLabels, List<LocalPSWorker> workers) {
-		// training data
-		List<MatrixObject> pfs = disjointContiguous(workers.size(), features);
-		List<MatrixObject> pls = disjointContiguous(workers.size(), labels);
+		List<MatrixObject> pfs = dp.doPartition(workers.size(), features);
+		List<MatrixObject> pls = dp.doPartition(workers.size(), labels);
 		if (pfs.size() < workers.size()) {
 			if (LOG.isWarnEnabled()) {
 				LOG.warn(String.format("There is only %d batches of data but has %d workers. "
-					+ "Hence, reset the number of workers with %d.", pfs.size(), workers.size(), pfs.size()));
+						+ "Hence, reset the number of workers with %d.", pfs.size(), workers.size(), pfs.size()));
```

Comment:
Minor nitpick (because I've seen this in previous PRs): the indentation for single line breaks is one not two tabs.